qtp_biom/validate.py

```diff
@@ -148,15 +149,18 @@ def validate(qclient, job_id, parameters, out_dir):
     # by SEPP for Deblur), if it exists
     tree = None
     if 'plain_text' in files:
-        phylogeny_fp = files['plain_text'][0]
-
-        try:
-            tree = bp.parse_newick(open(phylogeny_fp).read())
-            tree = bp.to_skbio_treenode(tree)
-            filepaths.append((phylogeny_fp, 'plain_text'))
-        except Exception:
-            return False, None, ("Phylogenetic tree cannot be parsed "
-                                 "via scikit-biom")
+        # first let's check if is a tgz, if it is, just pass the file
+        filename = files['plain_text'][0]
+        if is_tarfile(filename):
+            filepaths.append((filename, 'plain_text'))
```

Comment:
Is it okay that we're appending 'plain_text' along with a path to a tarball in this list? It's not clear to me whether the 'plain_text' is a reference to the file's uncompressed contents or the file itself. I'm concerned that the user of this list will have to know to exists() each filename to test if it's a tarball and process accordingly.

Reply:
Approved. This was discussed in qiita-admin.